images/ci-secret-generator/oc_sa_create_kubeconfig.sh

```diff
@@ -15,12 +15,56 @@ CLUSTER=$2
 readonly CLUSTER
 SERVICE_ACCOUNT=$3
 readonly SERVICE_ACCOUNT
+SA_NAMESPACE=$4
+readonly=SA_NAMESPACE
 
 TMP_KUBE_CONFIG_FILE="$(mktemp)"
 trap 'rm -rf ${TMP_KUBE_CONFIG_FILE}' EXIT
 
-
 URL=$(oc --kubeconfig "${CONFIG_UPDATER_DIR}/sa.config-updater.${CLUSTER}.config" config view -o jsonpath="{.clusters[0].cluster.server}")
-TOKEN=$(oc --kubeconfig "${CONFIG_UPDATER_DIR}/sa.config-updater.${CLUSTER}.config" create token -n ci ${SERVICE_ACCOUNT} --duration=2419200s)
-oc --kubeconfig "${TMP_KUBE_CONFIG_FILE}" login "${URL}" --token "${TOKEN}" > /dev/null
-cat "${TMP_KUBE_CONFIG_FILE}" | sed "s/${SERVICE_ACCOUNT}/${CLUSTER}/g"
+TOKEN=$(oc --kubeconfig "${CONFIG_UPDATER_DIR}/sa.config-updater.${CLUSTER}.config" create token -n ${SA_NAMESPACE} ${SERVICE_ACCOUNT} --duration=2419200s)
+
+INSECURE_SKIP_TLS_VERIFY="false"
+
+# vsphere uses a self signed cluster
+if [[ "${CLUSTER}" == "vsphere" ]]; then
+  INSECURE_SKIP_TLS_VERIFY="true"
+fi
+
+oc --kubeconfig "${TMP_KUBE_CONFIG_FILE}" login "${URL}" --token "${TOKEN}" --insecure-skip-tls-verify=${INSECURE_SKIP_TLS_VERIFY} > /dev/null
+
+CONTEXT_NAME=$(oc --kubeconfig "${TMP_KUBE_CONFIG_FILE}" config current-context)
+#It is required by Prow that the current context name is the cluster name
+oc --kubeconfig "${TMP_KUBE_CONFIG_FILE}" config rename-context ${CONTEXT_NAME} ${CLUSTER} > /dev/null
+#oc410 sa create-kubeconfig -h
```

Comment:
did you mean to leave this here?

Reply:
My intention was to justify the `config set-context --current --namespace=${SA_NAMESPACE}` cmd below.
It is not required by Prow but I want to stay close to the kubeconfig file generated by `oc sa create-kubeconfig`.
Otherwise, the namespace from `oc login`: The current project is `ci-op-2y5bt641`.

```
oc --kubeconfig /tmp/bot.b02.config config view
...
    namespace: ci-op-2y5bt641
    user: system:serviceaccount:ci:build02/api-build02-gcp-ci-openshift-org:6443
  name: ci-op-2y5bt641/api-build02-gcp-ci-openshift-org:6443/system:serviceaccount:ci:build02
current-context: ci-op-2y5bt641/api-build02-gcp-ci-openshift-org:6443/system:serviceaccount:ci:build02
...
```